### PR TITLE
fix: n+1 query when use search select

### LINKED ISSUE #271: Query n + 1 when use has_many and search_select
```
f.has_many :transitions, allow_destroy: true  do |t|
  t.input :task_definition_from_id, as: :search_select, url: task_definitions_admin_task_definitions_path
end
```

When i run this, its go again on database to fetch task_definition_from. But i have includes(:transitions), so I think its a bug.

I dont think should go to database and fetch the first, it can only fetch from relation belongs_to

```
    def selected_item
      @selected_item ||= selected_collection.first
    end
```

### PR DESCRIPTION
**Pull request by victorlcampos**
closes #271

### CODE REVIEW

File: app/inputs/search_select_input.rb [hunk 1]
```diff
@@ -11,6 +11,10 @@ def input_method
     valid_method
   end
 
+  def selected_item
+    object_class.reflect_on_association(association_name) ? object.send(association_name) : selected_collection.first
```
Comment (monkeyci) [inline app/inputs/search_select_input.rb]: Line is too long. [117/100]

Comment (cprodhomme) [inline app/inputs/search_select_input.rb]:
```suggestion
  return object.send(association_name) if object_class.reflect_on_association(association_name)

  selected_collection.first
```

**Comment by rjherrera** [comment]
Hi @victorlcampos ! Thanks for the PR, could you please rebase this branch with the latest changes from master (and maybe include the suggestion from @cprodhomme)? If you do we could review this PR this week, sorry for the long wait.

**Comment by difernandez** [comment]
From what I see, the associated issue #271 was marked as closed by #369, so I'm going to close this now. Thanks for the issue and PR!

**Pull request closed without merging**